src/avi/mode/command_line.clj

```diff
@@ -14,24 +15,32 @@
   [command]
   (every? #(Character/isDigit %) command))
 
+(defn- have-exclamation?
+  [command]
+  (string/includes? command "!"))
+
 (defn- command-fn
   [command-line]
   (ns-resolve 'avi.commands (symbol command-line)))
 
 (defn- process-command
-  [editor command-line]
-  (+> editor
-    (cond
-      (= "" command-line)
-      identity
-
-      (line-number? command-line)
-      (avi.commands/-NUMBER- command-line)
-
-      (command-fn command-line)
-      ((command-fn command-line))
-
-      :else
-      (assoc :message [:white :red (str ":" command-line " is not a thing")]))))
+  [editor command-line-raw]
+  (let [force? (have-exclamation? command-line-raw)
```

Comment:
Did you know that Clojure symbols can have `!` in them?

I would just define `w!` and `wq!` and so forth as separate commands in `avi.commands`.